lectures/dynamic_programming/wald_friedman.md

```diff
@@ -360,162 +360,181 @@ Later, doing this will help us obey **the don't repeat yourself (DRY)** golden r
 
 Let's code this problem up and solve it.
 
-We implement the cost functions for each choice considered in the
-Bellman equation {eq}`new3`.
+We discretize the belief space, set up quadrature rules for both
+likelihood distributions, and evaluate the Bellman operator on that
+grid. The helpers below encode the immediate losses, Bayes' rule for the
+posterior belief, and the expectation of the continuation value in
+{eq}`new4`.
 
-First, consider the cost associated to accepting either distribution and
-compare the minimum of the two to the expected benefit of drawing again.
+```{code-cell} julia
+accept_x0(p, L0) = (one(p) - p) * L0
+accept_x1(p, L1) = p * L1
+const BELIEF_FLOOR = 1e-6
 
-Drawing again will only be worthwhile if the expected marginal benefit of
-learning from an additional draw is greater than the explicit cost.
+clamp_belief(p) = clamp(float(p), BELIEF_FLOOR, one(Float64) - BELIEF_FLOOR)
 
-For every belief $p$, we can compute the difference between accepting a
-distribution and choosing to draw again.
+function gauss_jacobi_dist(F::Beta, N)
+    s, wj = FastGaussQuadrature.gaussjacobi(N, F.β - 1, F.α - 1)
+    x = (s .+ 1) ./ 2
+    C = 2.0^(-(F.α + F.β - 1.0)) / SpecialFunctions.beta(F.α, F.β)
+    return x, C .* wj
+end
 
-The solution $\alpha$, $\beta$ occurs at indifference points.
+function WFProblem(; d0 = Beta(1, 1), d1 = Beta(9, 9),
```

Comment:
We should be using lower case snake case for functions.  So call it `wf_problem` or something like that.